src/java/main/org/apache/zookeeper/server/quorum/Leader.java

```diff
@@ -584,6 +584,12 @@ void lead() throws IOException, InterruptedException {
                         }
                     }
 
+                    // check leader running status
+                    if (!this.isRunning()) {
+                        shutdown("Unexpected internal error");
```

Comment:
One question about this. Say we hit it here and print this message, how does the admin determines what was the actual cause of the error? Will the admin have to look for the exception that caused the problem? I'm just wondering about the troubleshooting path.